src/mpyl/cli/projects.py

```diff
@@ -159,5 +167,48 @@ def lint(obj: ProjectsContext, all_, extended):
         )
 
 
+@projects.command(help="Upgrade projects to conform with the latest schema")
+@click.option(
+    "--apply",
+    "-a",
+    is_flag=True,
+    help="Apply upgrade operations to the project files",
+)
+@click.pass_obj
+def upgrade(obj: ProjectsContext, apply):
+    paths = map(Path, _find_project_paths(True, obj.cli.repo, ""))
+    candidates = check_upgrades_needed(list(paths))
+    if not apply:
+        upgradable = check_upgrade(obj.cli.console, candidates)
+        number_in_need_of_upgrade = len(upgradable)
+        if number_in_need_of_upgrade > 0:
+            obj.cli.console.print(
+                f"{number_in_need_of_upgrade} projects need to be upgraded"
+            )
+            sys.exit(1)
+
+    with obj.cli.console.status("Checking for upgrades...") as status:
+        materialized = list(candidates)
+        need_upgrade = [path for path, diff in materialized if diff is not None]
+        number_of_upgrades = len(need_upgrade)
+        status.console.print(
+            f"Found {len(materialized)} projects, of which {number_of_upgrades} need to be upgraded"
+        )
+        status.stop()
+        if number_of_upgrades > 0 and Confirm.ask("Upgrade all?"):
+            status.start()
+            for path in need_upgrade:
+                status.update(f"Upgrading {path}")
+                upgraded = upgrade_file(path, UPGRADERS)
+                if upgraded:
+                    path.write_text(upgraded)
+            status.stop()
+            status.console.print(
+                Markdown(
+                    f"Upgraded {number_of_upgrades} projects. Validate with `mpyl projects lint`"
```

Comment:
```suggestion
                    f"Upgraded {number_of_upgrades} projects. Validate with `mpyl projects lint --all --extended`"
```

Reply:
Aren't we getting rid of the `--all` flag

Reply:
[[TECH-556] Suggest to do extend validation](https://github.com/Vandebron/mpyl/pull/207/commits/7a0b0ff87ebd680f57912c33c5c41299f46bec5c)